### ISSUE TITLE
CodeGen Record structs cause TypeLoadException

### ISSUE THREAD

**Issue by colethecoder** [louthy/language-ext#692]
Using this code:

```cs
[Record]
public partial struct Person
{
    public readonly string Forename;
    public readonly string Surname;
}
```

The generated code includes:

```cs
public static readonly Lens<Person, string> forename = Lens<Person, string>.New(_x => _x.Forename, _x => _y => _y.With(Forename: _x));
public static readonly Lens<Person, string> surname = Lens<Person, string>.New(_x => _x.Surname, _x => _y => _y.With(Surname: _x));
```

This compiles but at runtime it generates a TypeLoadException:

```
System.TypeLoadException: 'Could not load type 'ConsoleApp48.Person' from assembly 'ConsoleApp48, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.'
```

I believe this is due to the Lens referencing the struct itself and, because it is static, the JIT is unaware of the type at the point that it attempts to initialise the static value.

**Comment by louthy** [comment]
I have a fix for this that generates an embedded `class` within the `struct`:
```c#
    public partial struct Person : System.IEquatable<Person>, System.IComparable<Person>, System.IComparable
    {
        ...
        ..
        .

        public static Lens<Person, string> forename => __LensFields.forename;
        public static Lens<Person, string> surname => __LensFields.surname;
        static class __LensFields
        {
            public static readonly Lens<Person, string> forename = Lens<Person, string>.New(_x => _x.Forename, _x => _y => _y.With(Forename: _x));
            public static readonly Lens<Person, string> surname = Lens<Person, string>.New(_x => _x.Surname, _x => _y => _y.With(Surname: _x));
        }
    }
```
That allows for the lenses to still be initialised once, but accessed via a property on the `struct`.

I'll try and get it deployed this week.

**Comment by louthy** [comment]
Released in [`3.3.49`](https://github.com/louthy/language-ext/releases/tag/v3.3.49)
